bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/MqttChannelTypeProvider.java

```diff
@@ -40,11 +48,23 @@
  *
  */
 @NonNullByDefault
-@Component(immediate = false, service = { ChannelTypeProvider.class, ChannelGroupTypeProvider.class,
-        MqttChannelTypeProvider.class })
-public class MqttChannelTypeProvider implements ChannelGroupTypeProvider, ChannelTypeProvider {
+@Component(immediate = false, service = { ThingTypeProvider.class, ChannelTypeProvider.class,
+        ChannelGroupTypeProvider.class, MqttChannelTypeProvider.class })
+public class MqttChannelTypeProvider implements ThingTypeProvider, ChannelGroupTypeProvider, ChannelTypeProvider {
+    private @NonNullByDefault({}) ThingTypeRegistry typeRegistry;
```

Comment:
If you use constructor injection, you can make this final and omit the null-annotation. Also the setter and getter can be removed.